// data-analysis-python\src\get_all_tickers\get_all_tickers.proto
syntax = "proto3";

option go_package = "api-go/src/service/ms_gateway/get_all_tickers";

// 全世界の株式ティッカーを取得するためのサービスを定義
service GetAllTickersService {
  rpc GetAllTickers (GetAllTickersRequest) returns (GetAllTickersResponse);
}

// リクエストメッセージ（パラメータが不要の場合でも定義）
message GetAllTickersRequest {}

// 株式ティッカーを格納するレスポンスメッセージ
message GetAllTickersResponse {
  repeated string tickers = 1;  // ティッカーシンボルのリスト
}


// python用コンパイルコマンド /data-analysis-pythonにて
// python -m grpc_tools.protoc -I src/get_all_tickers --python_out=src/get_all_tickers --grpc_python_out=src/get_all_tickers src/get_all_tickers/get_all_tickers.proto


// Go用コンパイルコマンド /api-goにて
// protoc --proto_path=../data-analysis-python/src/get_all_tickers --go_out=./src/service/ms_gateway/get_all_tickers --go_opt=paths=source_relative --go-grpc_out=./src/service/ms_gateway/get_all_tickers --go-grpc_opt=paths=source_relative ../data-analysis-python/src/get_all_tickers/get_all_tickers.proto
